### ISSUE TITLE
DA-GAN: Instance-level Image Translation by Deep Attention Generative Adversarial Networks

### ISSUE THREAD

**Issue by wolegechu** [wolegechu/learning-every-day#2]
## Abstract
- Classic problem
  - set-level constraints cannot learn the instance-level correspondences
  - false positives (e.g. geometric or semantic artifacts)
- Proposed method
  - translate samples from two sets into translating instances in a **highly-structured latent space**
  - Deep attention encoder 
  - **Instance-level correspondences**: consequently discovered through attending on the learned instance pairs

## Details
- Introduction
  - goal: seek an algorithm which is capable of finding meaningful correspondences from both set-level and instance-level under *unsupervised* setting
  - challenge
    - [1] how to exploit instance-level constraints while lacking correct pairing relationship
    - [2] how to further strengthen the constraints on set-level to mitigate mode collapse 
- Contributions
  - [Architecture] Use *Deep Attention Encoder (DAE)* attending on instance-level and meaningful correspondences of samples.
![be2d96e0-583a-4305-8b72-6f8fd21216bb](https://user-images.githubusercontent.com/15357902/37771914-0399136a-2e14-11e8-8aee-6faf433e6020.png)
    - use sigmoid function approximate step function (for attention mask generation)
    - generator consists of several residual blocks, and the instance-level representations are concatenated along the channel dimension and fed into it.
  - [Loss]
    - instance-level: *consistency loss*  and *symmetry loss* to require the translated samples correspond to correct (can be considered as an auto-encoder type of loss)
    - set-level: Vanilla GAN loss
- Experiments
![4ffc42f5-ed77-4fa7-b0ec-f8769f0471cd](https://user-images.githubusercontent.com/15357902/37773450-9e17c20c-2e18-11e8-944b-e4f4cba1cbe0.png)
  - this is the visualized distribution of 10 classes of birds. Each color represents a birds class. Black crosses represent the distribution of the generated samples. (a): generated data distribution of DA-GAN. (b) generated data distribution of StackGAN
![qq20180322-211635](https://user-images.githubusercontent.com/15357902/37772675-52c46c30-2e16-11e8-8af7-9c2cf93bd7d2.png)
  - Domain Adaption: state-of-the-art
  - Text to Image Synthesis: much improvement in both Inception score (1.9~2.7) and #miss (17~70)
  - Object Transfiguration: outperform in both classification accuracy and realism
  - [falure cases] incorrect attention due to the weakly supervised learning mechanism
- Training
  - up-sampling block: nearest-neighbor upsampling followed by a 3x3 stride 1 convolution
  - Batch normalization and ReLU
  - Adam with 0.0002

## Personal Thoughts
- Use the sigmoid attention for region selecting is a good method that not only get the goodness of hard attention and also can be trained through bp
- Applications:
  - make universal image generation compositional
  - semantic-level feature alignment in task-specific domain adaption task or common subspace learning
